src/es_archiver/lib/indices/create_index_stream.js

```diff
@@ -41,9 +41,6 @@ export function createCreateIndexStream({ client, stats, skipExisting, log }) {
 
   async function handleIndex(record) {
     const { index, settings, mappings, aliases } = record.value;
-
-    // Determine if the mapping belongs to a pre-7.0 instance, for BWC tests, mainly
-    const isPre7Mapping = !!mappings && Object.keys(mappings).length > 0 && !mappings.properties;
```

Comment:
There's a huge amount of technical debt hiding under this one line 😞 

We'll need to find all `es_archives/**/mappings.json` which don't have a `value.mappings.properties` key and fix the mappings file to be compatible with 7.x without the `include_type_name` bwc escape hatch.

Reply:
E.g. https://github.com/elastic/kibana/blob/afe82e2054f7768139546bbd15c485265f833ee6/x-pack/test/functional/es_archives/auditbeat/default/mappings.json#L356-L361

Reply:
- [ ] x-pack/test/functional/es_archives/auditbeat/default/mappings.json
 - [ ] x-pack/test/functional/es_archives/dashboard/feature_controls/security/mappings.json
 - [ ] x-pack/test/functional/es_archives/dashboard/feature_controls/spaces/mappings.json
 - [ ] x-pack/test/functional/es_archives/discover/feature_controls/security/mappings.json
 - [ ] x-pack/test/functional/es_archives/discover/feature_controls/spaces/mappings.json
 - [ ] x-pack/test/functional/es_archives/infra/6.6.0/docker/mappings.json
 - [ ] x-pack/test/functional/es_archives/infra/7.0.0/hosts/mappings.json
 - [ ] x-pack/test/functional/es_archives/infra/legacy/mappings.json
 - [ ] x-pack/test/functional/es_archives/monitoring/apm/mappings.json
 - [ ] x-pack/test/functional/es_archives/monitoring/basic_6.3.x/mappings.json
 - [ ] x-pack/test/functional/es_archives/monitoring/beats-with-restarted-instance/mappings.json
 - [ ] x-pack/test/functional/es_archives/monitoring/beats/mappings.json
 - [ ] x-pack/test/functional/es_archives/monitoring/ccr/mappings.json
 - [ ] x-pack/test/functional/es_archives/monitoring/logstash-pipelines/mappings.json
 - [ ] x-pack/test/functional/es_archives/monitoring/multi-basic/mappings.json
 - [ ] x-pack/test/functional/es_archives/monitoring/multicluster/mappings.json
 - [ ] x-pack/test/functional/es_archives/monitoring/singlecluster-basic-beats/mappings.json
 - [ ] x-pack/test/functional/es_archives/monitoring/singlecluster-green-gold/mappings.json
 - [ ] x-pack/test/functional/es_archives/monitoring/singlecluster-green-platinum/mappings.json
 - [ ] x-pack/test/functional/es_archives/monitoring/singlecluster-green-trial-two-nodes-one-cgrouped/mappings.json
 - [ ] x-pack/test/functional/es_archives/monitoring/singlecluster-red-platinum/mappings.json
 - [ ] x-pack/test/functional/es_archives/monitoring/singlecluster-three-nodes-shard-relocation/mappings.json
 - [ ] x-pack/test/functional/es_archives/monitoring/singlecluster-yellow-basic/mappings.json
 - [ ] x-pack/test/functional/es_archives/monitoring/singlecluster-yellow-platinum--with-10-alerts/mappings.json
 - [ ] x-pack/test/functional/es_archives/monitoring/singlecluster-yellow-platinum/mappings.json
 - [ ] x-pack/test/functional/es_archives/monitoring/standalone_cluster/mappings.json
 - [ ] x-pack/test/functional/es_archives/spaces/copy_saved_objects/mappings.json
 - [ ] x-pack/test/functional/es_archives/spaces/disabled_features/mappings.json
 - [ ] x-pack/test/functional/es_archives/timelion/feature_controls/mappings.json
 - [ ] x-pack/test/functional/es_archives/uptime/pings/mappings.json
 - [ ] x-pack/test/functional/es_archives/visualize/default/mappings.json
 - [ ] x-pack/test/reporting/es_archives/bwc/6_2/mappings.json
 - [ ] x-pack/test/reporting/es_archives/bwc/6_3/mappings.json